### ISSUE TITLE
SHOC/fft dynamic trace issue

### ISSUE THREAD

**Issue by gaurav-1993singh** [harvard-acc/gem5-aladdin#13]
Hi,
I was trying to run the fft test in SHOC. In the normal mode (without dma) it makes the dynamic trace but it does not give out the power and area summary (it does give out the stats). When I try to make the trace using the dma-trace-binary, the fft-instrumented has a segmentation fault. Please let me know if there is a workaround. My main goal is to get power and area numbers.

Thanks

**Comment by xyzsam** [comment]
Thanks, I can reproduce. Investigating.

**Comment by gaurav-1993singh** [comment]
I will tell you my findings to hopefully save some of your time. The dmaLoad and dmaStore functions in the *.c files (testcases) are still on DMA_INTERFACE_V1, whereas the Makefile.tracer expect it to be on V3. Also, these files don't have the maparraytoaccelerator and invokeaccelerator commands. I think the makefiles should also automatically include the aladdin_sys_connection.c and aladdin_sys_constants.c files. I was able to get the power numbers for FFT by making all these hacks. In all, I think the .c files for test cases need update.

**Comment by xyzsam** [comment]
Yes, the discrepancy in expected DMA interface versions is the reason. We deprecated v1 but we missed updating most of these benchmarks (only the ones in our integration tests were fixed). Glad you were able to get the numbers you wanted. We'll get this issue fixed.
